src/sharing/SharingScreen.js

```diff
@@ -0,0 +1,467 @@
+/* @flow strict-local */
+import React, { PureComponent } from 'react';
+import { View, StyleSheet, Image, ScrollView, Modal, BackHandler } from 'react-native';
+import type { NavigationScreenProp } from 'react-navigation';
+import { createMaterialTopTabNavigator } from 'react-navigation';
+import { BRAND_COLOR } from '../styles/constants';
+import type { Dispatch, SharedData, Subscription, User, Auth } from '../types';
+import { connect } from '../react-redux';
+import { ZulipButton, Input, Label, Screen } from '../common';
+import UserItem from '../users/UserItem';
+import { getSubscriptionsById } from '../subscriptions/subscriptionSelectors';
+import { getUsersById } from '../users/userSelectors';
+import StreamAutocomplete from '../autocomplete/StreamAutocomplete';
+import TopicAutocomplete from '../autocomplete/TopicAutocomplete';
+import AnimatedScaleComponent from '../animation/AnimatedScaleComponent';
+import ChooseRecipientsScreen from './ChooseRecipientsScreen';
+import { sendMessage, uploadFile } from '../api';
+import { streamNarrow } from '../utils/narrow';
+import { getAuth } from '../selectors';
+import { showToast } from '../utils/info';
+import { navigateBack } from '../nav/navActions';
+import { fetchTopicsForActiveStream } from '../topics/topicActions';
+
+const styles = StyleSheet.create({
+  wrapper: {
+    flex: 1,
+    padding: 10,
+  },
+  imagePreview: {
+    margin: 10,
+    borderRadius: 5,
+    width: 200,
+    height: 200,
+  },
+  container: {
+    flex: 1,
+  },
+  actions: {
+    flexDirection: 'row',
+  },
+  button: {
+    flex: 1,
+    margin: 8,
+  },
+  usersPreview: {
+    padding: 10,
+  },
+  sharingWith: {
+    fontSize: 16,
+    marginBottom: 10,
+  },
+  chooseButton: {
+    marginTop: 8,
+    marginBottom: 8,
+    width: '50%',
+    alignSelf: 'flex-end',
+  },
+  message: {
+    height: 70,
+    borderRadius: 5,
+    borderWidth: 1,
+    borderColor: 'whitesmoke',
+    padding: 5,
+  },
+});
+
+type StreamScreenProps = $ReadOnly<{|
+  share: SharedData,
+  dispatch: Dispatch,
+  subscriptions: Map<number, Subscription>,
+  auth: Auth,
+|}>;
+
+type StreamScreenState = $ReadOnly<{|
+  stream: string,
+  topic: string,
+  message: string,
+  isStreamFocused: boolean,
+  isTopicFocused: boolean,
+|}>;
+
+class StreamScreen extends React.Component<StreamScreenProps, StreamScreenState> {
```

Comment:
This name is confusing because we already have a `StreamScreen`. The computer doesn't get confused, because this isn't exported, but us human readers (and readers of things like stack traces in the future!) might be :slightly_smiling_face: 

This one is really somewhat more specific. Perhaps `StreamShareScreen`?

Also, though, it's kind of odd calling this a "screen" when it's inside something else that is a "screen". I think we already have a number of examples of that oddity in the codebase already, but perhaps we can avoid adding more. :wink: How about just `StreamShare`? Or still more explicitly: `ShareToStream`?